src/native/libs/System.Globalization.Native/pal_collation.m

```diff
@@ -78,6 +79,126 @@ static NSStringCompareOptions ConvertFromCompareOptionsToNSStringCompareOptions(
     return mutableString;
 }
 
+/**
+ * Removes zero-width and other weightless characters such as U+200B (Zero Width Space), 
+ * U+200C (Zero Width Non-Joiner), U+200D (Zero Width Joiner), U+FEFF (Zero Width No-Break Space), 
+ * and the NUL character from the specified string.
+ */
+static NSString* RemoveWeightlessCharacters(NSString* source)
+{
+    NSError *error = nil;
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"[\u200B-\u200D\uFEFF\0]" options:NSRegularExpressionCaseInsensitive error:&error];
+
+    if (error != nil)
+        return source;
+
+    NSString *modifiedString = [regex stringByReplacingMatchesInString:source options:0 range:NSMakeRange(0, [source length]) withTemplate:@""];
+
+    return modifiedString;
+}
+
+// Structure to hold the modified string and character mapping
+typedef struct {
+    NSString* modifiedString;
+    NSArray<NSNumber*>* mapping; // mapping[i] = original position j for character at position i in modifiedString
+} StringWithMapping;
+
+/**
+ * Removes symbols including whitespace, punctuation, currency symbols, the percent sign, 
+ * mathematical symbols, the ampersand, and similar characters from the specified string.
+ * This corresponds to .NET's CompareOptions.IgnoreSymbols behavior.
+ */
+static NSString* RemoveSymbols(NSString* source)
+{
+    NSError *error = nil;
+    // Unicode categories covered:
+    // \p{P} - Punctuation (all punctuation marks)
+    // \p{S} - Symbols (currency, mathematical, modifier, and other symbols)
+    // \p{Z} - Separators (space, line, and paragraph separators)
+    NSRegularExpression *regex = [NSRegularExpression regularExpressionWithPattern:@"[\\p{P}\\p{S}\\p{Z}]" options:NSRegularExpressionCaseInsensitive error:&error];
+
```

Comment:
would make sense to factor this to its own inlined method? I am seeing using more than once